### PR TITLE
feat: built search component UI

### PR DESCRIPTION
**Pull request by rajat-ws**
## Screenshots:

- Search Component UI

image

## Description

- added search component UI
- wrote tests for `search`  and `header` component

### CODE REVIEW

File: features/itunes/api/getTracks.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if (action.type === HYDRATE) {
+      return action.payload[reducerPath];
+    }
+  },
+  endpoints: builder => ({
+    fetchTracks: builder.query<TrackResponse, string>({
+      query: name => `search?term=${name}`,
+    }),
+  }),
+});
```
Comment (praveenkumar1798) [inline features/itunes/api/getTracks.ts:30]: Don't we require response transformers here ?

Comment (praveenkumar1798) [inline features/itunes/api/getTracks.ts:30]: response transformers missing. Don't we need it here ?

File: features/itunes/components/Header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { injectIntl } from "react-intl";
+import styled from "styled-components";
+
+const HeaderContainer = styled.div`
+  && {
+    display: flex;
+    justify-content: center;
+    width: 50%;
+    margin: 0 auto;
+    background: #3c4048;
```
Comment (praveenkumar1798) [inline features/itunes/components/Header/index.tsx]: use from colors file

File: features/itunes/components/SearchBox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+const SearchBoxContainer = styled.div`
+  && {
+    width: 50%;
+    display: flex;
+    justify-content: center;
+    margin: 0 auto;
+
+    .ant-card {
+      border: 1px solid #3c4048;
```
Comment (praveenkumar1798) [inline features/itunes/components/SearchBox/index.tsx]: use from colors file

**Review by praveenkumar1798 — APPROVED** [review]

File: features/itunes/components/Header/index.tsx (hunk 2 above)
Comment (rajat-ws) [reply features/itunes/components/Header/index.tsx]: Yes, sure. Will change accordingly.

File: features/itunes/components/SearchBox/index.tsx (hunk 3 above)
Comment (rajat-ws) [reply features/itunes/components/SearchBox/index.tsx]: Will update accordingly

File: features/itunes/components/Header/index.tsx (hunk 2 above)
Comment (rajat-ws) [reply features/itunes/components/Header/index.tsx]: Updated

File: features/itunes/components/SearchBox/index.tsx (hunk 3 above)
Comment (rajat-ws) [reply features/itunes/components/SearchBox/index.tsx]: Updated

**Pull request merged**